system/CodeIgniter.php

```diff
@@ -907,8 +907,9 @@ protected function startController()
 	 */
 	protected function createController()
 	{
-		$class = new $this->controller();
-		$class->initController($this->request, $this->response, Services::logger());
+		$controllerFactory = Services::controllerfactory($this->request, $this->response, Services::logger());
```

Comment:
can you make this as camelCase?

Reply:
Do you mean `Services::controllerFactory()`?

Reply:
Ah nevermind. I forgot services are checked using `strtolower`